mitodl/ocw-studio
# Custom URL tweaks

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #1370 
Closes #1369 
Closes #1368 

#### What's this PR do?
- Saves `Website.url_path` instead of `Website.name` to resource collections and course lists, migrates data for existing ones.
- Modifies the video transcript metadata to point to the correct url based on parent website url
- Includes a working link above the URL field when publishing a subsequent time, but before a production publish.
- Fixes the link in ocw-www publishing success/failure emails

#### How should this be manually tested?
- Create some resource collections and course-lists for your www site, include sites where the name and url_path differ, reopen them after editing, make sure they work as expected, and that the values saved to git are for the `url_path` and not `name`.  Also check that any pre-existing resource collections and course-lists have been updated to use `url_path` (the git metadata/markdown should show `courses/<path_url>` instead of just `<name>`)
- Follow test instructions for https://github.com/mitodl/ocw-studio/pull/1345, but at the end publish a couple times with different url paths and ensure the caption/transcript path metadata gets updated with the new path each time.
- Create a new video gallery for a site and ensure it still works.
- Create a new site, choose a url, publish to draft (it will fail, that's ok), then open the publish drawer again.  This time there should be a real link above the URL field.
- Test out with [this branch of ocw-hugo-themes](https://github.com/mitodl/ocw-hugo-themes/pull/700) to ensure everything renders properly with the new values for resource collections, video galleries, course lists, search, etc.


#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
<img width="706" alt="Screen Shot 2022-05-19 at 1 11 05 PM" src="https://user-images.githubusercontent.com/187676/169590577-240be5c3-7dd9-4bce-aeb9-fd6794190279.png">


#### What GIF best describes this PR or how it makes you feel?
(Optional)
